stdlib/test/builtin/test_simd.mojo

```diff
@@ -1008,6 +1008,53 @@ def test_indexer():
     assert_equal(0, Scalar[DType.bool](False).__index__())
 
 
+def test_reduce():
+    @parameter
+    def test_dtype[type: DType]():
+        @parameter
+        if type.is_numeric():
+            assert_equal(SIMD[type, 4](1, 2, 3, 4).reduce_add(), 10)
+            assert_equal(SIMD[type, 4](1, 2, 3, 4).reduce_mul(), 24)
+            assert_equal(SIMD[type, 4](1, 0, 3, 4).reduce_mul(), 0)
+            assert_equal(SIMD[type, 4](1, 2, 3, 4).reduce_min(), 1)
+            assert_equal(SIMD[type, 4](1, 2, 3, 4).reduce_max(), 4)
+
+        @parameter
+        if type.is_signed():
+            assert_equal(SIMD[type, 4](1, -2, 3, 4).reduce_add(), 6)
+            assert_equal(SIMD[type, 4](1, -2, 3, 4).reduce_mul(), -24)
+            assert_equal(SIMD[type, 4](1, -2, 3, 4).reduce_min(), -2)
+            assert_equal(SIMD[type, 4](1, -2, 3, 4).reduce_max(), 4)
+
+        @parameter
+        if type.is_bool():
+            assert_true(SIMD[type, 4](True, True, True, True).reduce_and())
+            assert_false(SIMD[type, 4](True, False, True, True).reduce_and())
+            assert_true(SIMD[type, 4](False, True, False, False).reduce_or())
+            assert_false(SIMD[type, 4](False, False, False, False).reduce_or())
+
+        @parameter
+        if type.is_integral():
+            assert_equal(SIMD[type, 4](2, 6, 10, 18).reduce_and(), 2)
+            assert_equal(SIMD[type, 4](1, 0, 3, 4).reduce_and(), 0)
+            assert_equal(SIMD[type, 4](1, 3, 4, 8).reduce_or(), 15)
+            assert_equal(SIMD[type, 4](0, 0, 6, 0).reduce_or(), 6)
+
+    test_dtype[DType.bool]()
+    test_dtype[DType.int8]()
+    test_dtype[DType.int16]()
+    test_dtype[DType.int32]()
+    test_dtype[DType.int64]()
+    test_dtype[DType.uint8]()
+    test_dtype[DType.uint16]()
+    test_dtype[DType.uint32]()
+    test_dtype[DType.uint64]()
+    test_dtype[DType.float16]()
```

Comment:
Could you please add a cast for `bfloat16` as well? You will have to guard it with
```mojo
@parameter
    if not has_neon():
```

Reply:
sure 🌻